File: insertion_sort/insertion_sort.go
package main

import (
    "fmt"
    "math/rand"
    "time"
)

// Sort array using the Insertion Sort technique
func insertion_sort(array []int) {
    for i := range array {
        swaped := false
        for j:=len(array)-1; j>i; j-- {
            if array[j] < array[j-1] {
                array[j], array[j-1] = array[j-1], array[j]
                swaped = true
            }
        }
        if !swaped { break }
    }
}

// The main function
func main() {
    rand.Seed(time.Now().UnixNano())
    arr := make([]int, 15);
    for i:=0; i<len(arr); i++ {
        arr[i] = rand.Intn(99)
    }
    fmt.Println(arr);
    insertion_sort(arr);
    fmt.Println(arr);
}

File: insertion_sort/insertion_sort.py
""" Insertion Sort in Python
"""
import random

def insertion_sort(array:list):
    """ Order array using the Insertion Sort method """
    for i in range(len(array)):
        swaped = False
        for j in range(len(array)-1, i, -1):
            if array[j] < array[j-1]:
                array[j-1], array[j] = array[j], array[j-1]
                swaped = True
        if not swaped:
            break

if __name__ == "__main__":
    arr  = [random.randint(1, 99) for _ in range(12)]
    print(arr)
    insertion_sort(arr)
    print(arr)
